src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/ProxyHeaderModule.cs

```diff
@@ -80,6 +89,15 @@ public void UseHeaders(NameValueCollection requestHeaders, NameValueCollection s
             serverVariables.Set(ServerHttps, value.IsSecure ? On : Off);
 
             requestHeaders[Host] = host;
+
+            // Need to use reflection to force HttpRequest to update the known headers
+            if (request != null)
```

Comment:
This needs to be done on the code path for line 111

Reply:
Thanks for catching this